docs/release_notes.rst

```diff
@@ -2,6 +2,81 @@
 PUDL Release Notes
 =======================================================================================
 
+---------------------------------------------------------------------------------------
+v2023.XX.XX
+---------------------------------------------------------------------------------------
+
+Dagster Adoption
+^^^^^^^^^^^^^^^^
+* After comparing comparing python orchestration tools :issue:`1487`, we decided to
+  adopt `Dagster <https://dagster.io/>`__. Dagster will allow us to parallize the ETL,
+  persist datafarmes at any step in the data cleaning process, visualize data
+  depedencies and run subsets of the ETL from upstream caches.
+* We are converting PUDL code to use dagster concepts in two phases. The first phase
+  converts the ETL portion of the code base to use
+  `software defined assets <https://docs.dagster.io/concepts/assets/software-defined-assets>`__
+  :issue:`1570`. We will convert the pandas computations cached in the
+  :mod:`pudl.output.pudltabl.PudlTabl` class to use software defined assets in
+  phase 2 :issue:`1973`.
+* General changes:
+
+  * :mod:`pudl.etl` is now a subpackage that collects all pudl assets into a dagster
+    `Definition <https://docs.dagster.io/concepts/code-locations>`__.
+  * The ``pudl_settings``, ``Datastore`` and ``DatasetSettings`` are now dagster
+    resources. See :mod:`pudl.resources`.
+  * IO to sqlite is now handled using the
+    :mod:`pudl.io_managers.pudl_sqlite_io_manager`.
+    Removes the need for the ``pudl.load`` module.
+  * The ``pudl_etl``  and ``ferc_to_sqlite`` commands no longer support loading
+    specific tables. The commands run all of the tables. Use dagster assets to
+    run subsets of the tables.
+  * The ``--clobber`` argument has been removed from the ``pudl_etl`` command.
+  * New static method :mod:`pudl.metadata.classes.Package.get_etl_group_tables`
+    returns the resources ids for a given etl group.
+  * :mod:`pudl.settings.Ferc2XbrlToSqliteSettings` class now loads all FERC
+    datasources if no datasets are specified.
+
+* EIA ETL changes:
+
+  * EIA extract methods are now ``@multi_asset`` that return an asset for each
+    raw table. 860 and 923 are separate ``@multi_asset`` which allows this data
+    to be extracted in parallel.
+  * ``pudl.transform.eia860.transform()`` and ``pudl.transform.eia923.transform()``
+    functions have been deprecated. The EIA table level cleaning functions are now
+    dagster ``@asset``. The table level cleaning assets now have a "clean\_" prefix
+    and a "_{datasource}" suffix to distinguish them from the final harvested tables.
+  * ``pudl.transform.eia.transform()`` is now a ``@multi_asset`` that depends
```

Comment:
Let's try and use the `:mod:` and `:func:` etc. syntax across the board to create links to these entities in the API docs.

Reply:
I didn't use :mod: and :func: for the modules and functions that are removed in the `dagster-asset-etl` branch. I assumed sphinx will throw an error if the mod or func doesn't exist in the codebase.

Reply:
Ah okay.